includes/legacy/class-wc-gateway-amazon-payments-advanced-subscriptions-legacy.php

```diff
@@ -366,9 +377,21 @@ private function get_billing_agreement_details( $order_id, $amazon_billing_agree
 			)
 		);
 
-		$this->handle_generic_api_response_errors( __METHOD__, $response, $order_id, $amazon_billing_agreement_id );
+		try {
+			$this->handle_generic_api_response_errors( __METHOD__, $response, $order_id, $amazon_billing_agreement_id );
 
-		wc_apa()->log( "Info: GetBillingAgreementDetails for Billing Agreement ID: {$amazon_billing_agreement_id}." );
+			wc_apa()->log( "Info: GetBillingAgreementDetails for Billing Agreement ID: {$amazon_billing_agreement_id}." );
+		} catch ( Exception $e ) {
+			wc_apa()->log( "Error: Exception encountered in 'GetBillingAgreementDetails': {$e->getMessage()}" );
+
+			$return = new WP_Error();
+			$return->add(
```

Comment:
Why add instead of just instantiating a WP Error by using the construct? Do we need several errors here?